controllers/imagerepository_controller.go

```diff
@@ -474,14 +478,15 @@ func getURLHost(urlStr string) (string, error) {
 	// http://docker.io and http://docker.io/v1/, etc.
 	// So strip everything down to the host.
 	// Also, the registry might be local and on a different port.
-	// So we need to check for that because url.Parse won't parse the ip:port format very well.
 	u, err := url.Parse(urlStr)
 	if err != nil {
 		return "", err
 	}
 
 	if u.Host == "" {
-		return "", errors.New(fmt.Sprintf("Invalid registry auth url: %s. Must be a valid URI (e.g. https://gcr.io/v1/)", urlStr))
+		return "", errors.New(fmt.Sprintf(
+			"Invalid registry auth url: %s. Must be a valid hostame(e.g. gcr.io) or url (e.g. https://gcr.io/v1/)",
```

Comment:
Sorry to be a nitpicker -- it could also be a non-hostname, non-URL, like `index.docker.io/v2` and get through the func successfully. The requirement is that either it's a valid HTTP[S] URL, or adding https:// makes it a valid HTTPS URL.

Reply:
I realise lots of the things I'm commenting on actually come from the code you're borrowing, for which I can only apologise :-)

Reply:
Sure. I will correct that.

Reply:
Is this okay ?
```
Must be a valid http URL (e.g. https://gcr.io/v1/) or prepending with 'https://' makes it a valid http URL
```